### PR TITLE
[FEATURE] COCO-891 outbrain connector

### PR DESCRIPTION
**Pull request by AdamBkl**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/schemas/budgets.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                },
+                                "pacing": {
+                                    "type": "string"
+                                },
+                                "dailyTarget": {
+                                    "type": "integer"
+                                },
+                                "maximumAmount": {
+                                    "type": "integer"
+                                }
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/schemas/budgets.json:64]: Beware that floats will cause issues .. `number`

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/schemas/campaigns.json [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                        },
+                        "amountSpent": {
+                            "type": "integer"
+                        },
+                        "targeting": {
+                            "type": "object",
+                            "properties": {
+                                "platform": {
+                                    "type": "array",
+                                    "items": [
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/schemas/campaigns.json:55]: don't use a fixed number here please

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/spec.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "required": ["USERNAME", "PASSWORD", "MARKETER_ID","START_DATE"],
+    "additionalProperties": false,
+    "properties": {
+      "USERNAME": {
+        "type": "string",
+        "description": "Outbrain Username"
+      },
+      "PASSWORD": {
+        "type": "string",
+        "description": "Outbrain Password"
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/spec.json:16]: `"airbyte_secret": true`

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.sources.streams.http.auth import NoAuth
+import logging
+import datetime
+"""
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py:17]: Let's remove these comments please...

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        TODO: Replace the streams below with your own streams.
+
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        url = 'https://api.outbrain.com/amplify/v0.1/login'
+         
+        auth=requests.auth.HTTPBasicAuth(config["USERNAME"], config["PASSWORD"])
+        response = (requests.get(url, auth=auth))
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py:404]: You could reuse the `url` specified above

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        """
+        TODO: Override this method to define the path this stream corresponds to. E.g. if the url is https://example-api.com/v1/customers then this
+        should return "customers". Required.
+        """
+        marketer_id = self.marketer_id
+        return 'https://api.outbrain.com/amplify/v0.1/marketers/' +marketer_id + '/budgets?detachedOnly=false'
```
Comment (dinks) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py:125]: Why not a path here? I see this repeated below as well...

File: airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py [hunk 7]
```diff
@@ -300,10 +271,8 @@ def retry_factor(self) -> int:
         return 1800
 
     def check_connection(self, logger, config) -> Tuple[bool, any]:
-        url = 'https://api.outbrain.com/amplify/v0.1/login'
-    
         auth=requests.auth.HTTPBasicAuth(config["USERNAME"], config["PASSWORD"])
-        check_connection_respone = (requests.get(url, auth=auth))
+        check_connection_respone = (requests.get(self.url, auth=auth))
```
Comment (AdamBkl) [inline airbyte-integrations/connectors/source-outbrain-amplify-api/source_outbrain_amplify_api/source.py:275]: is this what you meant by reusing the url from above?

**Review by dinks — APPROVED** [review]
Apart from the discussed points, all good!

**Pull request merged**